### ISSUE TITLE
Add property based tests for the VotingPower calculator

### ISSUE THREAD

**Issue by shonfeder** [informalsystems/tendermint-rs#821]
Followup to #333 

In the exploratory work of #815, our focused application of PBT surfaced an edge case that could produce fatal errors. Consequently, in discussion with @thanethomson, we decided it was worth investing in further application of PBT, both to improve the depth of tests and to continue building up a library of combinators to expedite writing PBT of future development.

The proposed target for the next step of this work is the [`VotingPowerCalculator`](https://github.com/informalsystems/tendermint-rs/blob/master/light-client/src/operations/voting_power.rs#L105). 

This work item decomposes into two principle parts:

1. Defining combinators for generating arbitrary inputs to `voting_power_in`.
2. Identifying the essential properties of this computation that should be checked, and writing the tests.

**What's the definition of "done" for this issue?**

1. There are combinators for generating arbitrary values of the following types:
    -   [ ] **SignedHeader:** /src/block/signed_header.rs
        -   [ ] **Header:** /src/block/header.rs
            -   [x]  **Version:** /src/block/header.rs
            -   [x] **Height:** /src/block/height.rs
            -   [x] **Time:** /src/time.rs
            -   [ ] **Hash:** /src/hash.rs
            -   [ ] **AppHash:** /src/hash.rs
        -   [ ] **Commit:** /src/block/commit.rs
    -   [ ] **ValidatorSet:** /src/validator.rs
        -   [ ] **Info:** /src/validator.rs
            -   [ ] **ID:** /src/account.rs
            -   [ ] **PublicKey:** /src/public_key.rs
            -   [ ] **Power:** /src/vote/power.rs
            -   [ ] **ProposerPriority:** /src/validator.rs
        -   [ ] **TrustThresholdFraction:** /src/trust_threshold.rs
2. There are property based tests for essential properties of the calculation written, and these are tagged with the logical unit they are derived from.

**Comment by shonfeder** [comment]
Yesterday we talked about where these generators might live, given that:

- they will likely need to be used by tests that outside of the tendermint crate
- we don't want to bloat tendermint with code that is only used in tests.

@romac advised that the best course of action would be to put the generators in a model guarded by a disjunctive feature that's satisfied either by `test` or `proprest` (or whatever other name we'd like here) using https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-attribute. E.g., 

```rust
#[cfg(any(test, proptest)]
```

Tests that need these generators can then specify the `proptest` feature when they depend on the tendermint crate.

**Comment by romac** [comment]
Absolutely! Although I think we want `proptest` to be a feature rather than a config flag, so the attribute would look like:

```
#[cfg(any(test, feature = "proptest")]
```

**Comment by shonfeder** [comment]
Yes indeed. Thank you for the correction!
